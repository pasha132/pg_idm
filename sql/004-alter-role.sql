begin;

insert into idm.roles (rolname) values ('test_role_001');
update idm.roles set rolname = 'test_role_002' where rolname = 'test_role_001';

select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;

update idm.roles
set
    rolsuper = true,
    rolinherit = false,
    rolcreaterole = true,
    rolcreatedb = true,
    rolcanlogin = true,
    rolreplication = true,
    rolconnlimit = 10,
    rolvaliduntil = to_timestamp('2000-01-01', 'YYYY-MM-DD'),
    rolbypassrls = true
where rolname = 'test_role_002';

select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;

update idm.roles
set
    rolconnlimit = -1,
    rolvaliduntil = null
where rolname = 'test_role_002';

select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;

rollback;