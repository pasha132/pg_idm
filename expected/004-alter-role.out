begin;
insert into idm.roles (rolname) values ('test_role_001');
update idm.roles set rolname = 'test_role_002' where rolname = 'test_role_001';
select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;
    rolname    | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolconnlimit | rolvaliduntil | rolbypassrls 
---------------+----------+------------+---------------+-------------+-------------+----------------+--------------+---------------+--------------
 test_role_002 | f        | t          | f             | f           | f           | f              |           -1 |               | f
(1 row)

update idm.roles
set
    rolsuper = true,
    rolinherit = false,
    rolcreaterole = true,
    rolcreatedb = true,
    rolcanlogin = true,
    rolreplication = true,
    rolconnlimit = 10,
    rolvaliduntil = to_timestamp('2000-01-01', 'YYYY-MM-DD'),
    rolbypassrls = true
where rolname = 'test_role_002';
select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;
    rolname    | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolconnlimit | rolvaliduntil | rolbypassrls 
---------------+----------+------------+---------------+-------------+-------------+----------------+--------------+---------------+--------------
 test_role_002 | t        | f          | t             | t           | t           | t              |           10 | 2000-01-01    | t
(1 row)

update idm.roles
set
    rolconnlimit = -1,
    rolvaliduntil = null
where rolname = 'test_role_002';
select
    rolname,
    rolsuper,
    rolinherit,
    rolcreaterole,
    rolcreatedb,
    rolcanlogin,
    rolreplication,
    rolconnlimit,
    to_char(rolvaliduntil, 'YYYY-MM-DD') as rolvaliduntil,
    rolbypassrls
from idm.roles
where rolname like 'test_role_%'
order by rolname;
    rolname    | rolsuper | rolinherit | rolcreaterole | rolcreatedb | rolcanlogin | rolreplication | rolconnlimit | rolvaliduntil | rolbypassrls 
---------------+----------+------------+---------------+-------------+-------------+----------------+--------------+---------------+--------------
 test_role_002 | t        | f          | t             | t           | t           | t              |           -1 |               | t
(1 row)

rollback;
